ngraph/core/include/ngraph/op/constant.hpp

```diff
@@ -383,80 +553,110 @@ namespace ngraph
+                    write_to_buffer(values);
+                }
+
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type != element::Type_t::u1 &&
+                                                      Type != element::Type_t::u4 &&
+                                                      Type != element::Type_t::i4,
+                                                  bool>::type = true>
+                void write_buffer(const std::vector<T>& source)
+                {
+                    auto p = get_data_ptr_nc<Type>();
+                    for (size_t i = 0; i < source.size(); i++)
+                    {
+                        p[i] = static_cast<DataStorageType>(source[i]);
+                    }
                 }
 
-                template <typename T, typename U>
-                static void write_buffer(void* target, const std::vector<U>& source, size_t count)
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type == element::Type_t::u4 ||
+                                                      Type == element::Type_t::i4,
+                                                  bool>::type = true>
+                void write_buffer(const std::vector<T>& source)
                 {
-                    T* p = reinterpret_cast<T*>(target);
-                    for (size_t i = 0; i < count; i++)
+                    auto p = get_data_ptr_nc<Type>();
+                    size_t i = 0;
+                    for (; i < source.size() / 2; i++)
+                    {
+                        const auto v1 = value_in_range<Type>(source[i * 2]) & 0x0F;
+                        const auto v2 = value_in_range<Type>(source[i * 2 + 1]) & 0x0F;
+                        const auto v = (v1 << 4) | v2;
+                        p[i] = static_cast<DataStorageType>(v);
+                    }
+                    if (source.size() % 2)
                     {
-                        p[i] = static_cast<T>(source[i]);
+                        const auto v1 = value_in_range<Type>(source[i * 2]) & 0x0F;
+                        const auto v = v1 << 4;
+                        p[i] = static_cast<DataStorageType>(v);
                     }
                 }
 
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type == element::Type_t::u1, bool>::type = true>
+                void write_buffer(const std::vector<T>& source)
+                {
+                    auto p = get_data_ptr_nc<Type>();
```

Comment:
Can we add `value_in_range` for `u1`?

Reply:
I'm not sure if we have to do it when I look on conversion from string https://github.com/openvinotoolkit/openvino/pull/5311/files#diff-21495300b7eb66fbb15c88efb4631ea25230e391293c9e006478012d5d1ac1b3R74